python/tests/test_reward.py

```diff
@@ -10,6 +10,13 @@ def test_IsDone(state):
     assert reward_func.get(state, done=True) == 1
 
 
+def test_NLPIterations(state):
+    reward_func = R.NLPIterations()
+    reward_func.reset(state)
+    reward_func.get(state)
+    assert reward_func.get(state, done=True) == 0
```

Comment:
> That's actually interesting! What is the behavior of a reward function when called twice on the same state?
How the library is build, reward functions are called on the environment side. Users should never call rewards functions by themselves. Then we can just decide that rewards functions are meant to be called once and only once per state, and behaviour is undefined otherwise.
> I guess it does not have to be the same. An environment could decide not to change a state, but that would still be a transition.
I guess we can always make that clear by adding something like `state_id` to distinguish between two identical states s_i and s_i+1.
> 
> Maybe we should change the name of the method (`is_done`) that sounds too much like a getter, but a getter would be consistent. Maybe `compute_reward`?
I don't really get that point. Where is that `is_done` method ? I agree, `compute` is more what the purpose of a reward function is. It computes the reward. Sometimes the computation is trivial, sometimes it will be more tricky (I think about dual gap integral for example).

Reply:
> I would remove nonetheless remove `done=True` as I find it obscure the intent of your test.

I disagree. It tests rather the reward function works fine on a terminal state.

Reply:
> I guess we can always make that clear by adding something like state_id to distinguish between two identical states s_i and s_i+1.

I like that, it makes it more intuitive and easy to debug. I would just open an issue for now.

Reply:
>> I would remove nonetheless remove done=True as I find it obscure the intent of your test.

> I disagree. It tests rather the reward function works fine on a terminal state.

We can leave like this, it is not a important. My point was the value `0` doesn't come from the fact that it is terminal, it comes from the fact that it is called twice. So I would personally go with three calls to `get`
```python
reward_func.get(state) 
assert reward_func.reset(state) == 0  # Check 0 if no LP iteration increase
reward_func.get(state, done=True)  # Check works with different param, no return assumed
```